mlir/include/mlir/Dialect/Utils/IndexingUtils.h

```diff
@@ -40,7 +40,7 @@ class ArrayAttr;
 /// Assuming `sizes` is `[s0, .. sn]`, return the vector<int64_t>
 ///   `[s1 * ... * sn, s2 * ... * sn, ..., sn, 1]`.
 ///
-/// `sizes` elements are asserted to be non-negative.
+/// `sizes` elements `s1` to `sn` are asserted to be non-negative.
```

Comment:
Is this relaxation of the assert needed in this PR?

Reply:
Yes, it's needed because the leading dimension can appear dynamic.